### PR TITLE
provider/aws: Add Lightsail Instance

### PR DESCRIPTION
**Pull request by catsby**
Support for [AWS Lightsail](https://lightsail.aws.amazon.com/ls/docs/all)

Includes:

- [x] Docs for `aws_lightsail_instance`
- [x] Tests for `aws_lightsail_instance`
- [x] `aws_lightsail_instance` resource

### CODE REVIEW

**Comment by catsby** [comment]
We're limited on `key_pair` right now, we'll need a separate resource for that

File: builtin/providers/aws/resource_aws_lightsail_instance.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+		BlueprintId:      aws.String(d.Get("blueprint_id").(string)),
+		BundleId:         aws.String(d.Get("bundle_id").(string)),
+		InstanceNames:    aws.StringSlice([]string{iName}),
+	}
+
+	if v, ok := d.GetOk("key_pair_name"); ok {
+		req.KeyPairName = aws.String(v.(string))
+	}
+	// // encoding required?
+	// if v, ok := d.GetOk("user_data"); ok {
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_lightsail_instance.go]: This intentionally missing? :)

File: builtin/providers/aws/resource_aws_lightsail_instance.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		// We don't return an error here because the Create call succeded
+		log.Printf("[ERR] Error waiting for instance (%s) to become ready: %s", d.Id(), err)
+	}
+
+	// return resourceAwsLightsailInstanceUpdate(d, meta)
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_lightsail_instance.go]: worth removing this old comment?

Comment (catsby) [inline builtin/providers/aws/resource_aws_lightsail_instance.go]: whoops :)

**Comment by catsby** [comment]
@stack72 thanks for the review, I've updated it!

> 1 other thing, is it worth adding a few more tests? This would allow us to check for the _disappears

I added `TestAccAWSLightsailInstance_disapear` for this 

> import

The `_basic` test has the `IDRefreshName` thing, any more than that?

> maybe an update

Sadly everything is `ForceNew` at the moment, the API doesn't offer much for updates at all. The resource lacks an `Update` method

**Comment by stack72** [comment]
LGTM! thanks for the changes :)

**Pull request merged**